File: SOEN2030 数据结构课程实践/实验/2021-09-30 实验四/Exercise1.java
public class Exercise1 {

    public static void main(String[] args) {
        int n = 4;
        int[] c = new int[n * 2 - 1];
        for (int i = 0; i < n; i++) {
            c[i] = i + 1;
        }
        for (int i = n; i < c.length; i++) {
            c[i] = c[i - 1] - 1;
        }
        for (int k = 0; k < c.length; k++) {
            int i = c[k];
            for (int j = 0; j < (n - i) * 2; j++) {
                System.out.print(" ");
            }
            for (int j = 0; j < i; j++) {
                System.out.print("*");
            }
            for (int j = 0; j < (i - 1) * 2; j++) {
                System.out.print(" ");
            }
            for (int j = 0; j < i; j++) {
                System.out.print("*");
            }
            System.out.println();
        }
    }

}

File: COMS3040 信息检索综合实践/实验/2023-03-08 实验3 页面信息抽取/webpage-information-extraction/src/exercise1.ts
import fs from 'node:fs/promises';

import * as cheerio from 'cheerio';

import { extractText } from './utils.js';

/**
 * Process a html file and save the result to another file.
 * @param sourcePath
 * @param outputPath
 * @param logOutput
 */
const exercise1 = async (
  sourcePath: string,
  outputPath: string,
  logOutput = false,
) => {
  const html = await fs.readFile(sourcePath, 'utf-8');
  const $ = cheerio.load(html);

  // Title
  const title = $('title').text();
  fs.writeFile(outputPath, `title:\n${title}\n\n`, 'utf-8');
  if (logOutput) {
    console.log(`title:\n${title}\n`);
  }

  // Body
  const body = $('body')
    .contents()
    .map((_, el) => extractText($, el)) // Extract text from each node.
    .get()
    .join('')
    .replace(/\n{2,}/g, '\n') // Remove redundant newlines.
    .trim()
    .split('Google Scholar', 2)[0]; // Remove words after "Google Scholar".
  await fs.appendFile(outputPath, `body:\n${body}\n`, 'utf-8');
  if (logOutput) {
    console.log(`body:\n${body}`);
  }
};

export default exercise1;
